CPAC/pipeline/random_state/seed.py

```diff
@@ -0,0 +1,188 @@
+
+    Examples
+    --------
+    >>> list(random_seed_flags().keys())
+    ['functions', 'interfaces']
+    >>> all([isinstance(random_seed_flags()[key], dict) for key in [
+    ...     'functions', 'interfaces']])
+    True
+    >>> set_up_random_state('random')
+    >>> list(random_seed_flags().keys())
+    ['functions', 'interfaces']
+    >>> all([isinstance(random_seed_flags()[key], dict) for key in [
+    ...     'functions', 'interfaces']])
+    True
+    '''
+    seed = random_seed()
+    if seed is None:
+        return {'functions': {}, 'interfaces': {}}
+    return {
+        'functions': {
+            # function: lambda function to apply to function source
+            hardcoded_reg: lambda fn_string: fn_string.replace(
+                'regcmd = ["antsRegistration"]',
+                f'regcmd = ["antsRegistration", "--random-seed", \"{seed}\"]'
+            )
+        },
+        'interfaces': {
+            # interface: [flags to apply]
+            # OR
+            # interface: ([flags to apply], [flags to remove])
+            #
+            # ANTs
+            # NOTE: Atropos gives the option "Initialize internal random number
+            #       generator with a random seed. Otherwise, initialize with a
+            #       constant seed number," so for Atropos nodes, the built-in
+            #       Atropos constant seed is used if a seed is specified for
+            #       C-PAC
+            AI: _reusable_flags()['ANTs'],
+            Registration: _reusable_flags()['ANTs'],
+            Atropos: (['--use-random-seed 0'],
+                      [flag for one in ['', ' 1'] for flag in
+                      [f'--use-random-seed{one}', f'-r{one}']]),
+            # FreeSurfer
+            ReconAll: ['-norandomness', f'-rng-seed {seed}'],
+            ApplyVolTransform: _reusable_flags()['FSL'],
+            # FSL
+            ImageMaths: _reusable_flags()['FSL'],
+            MathsCommand: _reusable_flags()['FSL']
+        }
+    }
+
+
+def _reusable_flags():
+    seed = random_seed()
+    return {
+        'ANTs': [f'--random-seed {seed}'],
+        'FSL': [f'-seed {seed}']
+    }
+
+
+def set_up_random_state(seed, log_dir=None):
+    '''Prepare C-PAC for random seed setting and logging
+
+    Parameters
+    ----------
+    seed : int, 'random', or None
+
+    log_dir : str, optional
+
+    Returns
+    -------
+    workflow
+
+    Examples
+    --------
+    >>> from CPAC.pipeline.random_state import random_seed
+    >>> set_up_random_state('random')
+    >>> isinstance(random_seed(), int)
+    True
+    >>> set_up_random_state('rando')
+    Traceback (most recent call last):
+    ValueError: Valid random seeds are positive integers up to 2147483647, "random", or None, not rando
+    >>> set_up_random_state(100)
+    >>> random_seed()
+    100
+    >>> set_up_random_state(0)
+    Traceback (most recent call last):
+    ValueError: Valid random seeds are positive integers up to 2147483647, "random", or None, not 0
```

Comment:
What does "None" correspond to? No specification of random state, so it is truly random + not logged?

Reply:
Yeah, `None` is the setting to have C-PAC perform the same behavior as it did before this PR